swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/AbstractResourceHandler.kt

```diff
@@ -144,12 +147,50 @@ abstract class AbstractResourceHandler<out T : Resource>(
           }
         }
     } catch (e: Exception) {
-      log.error("failed to cleanup resource", markedResource, e)
+      recordFailureForAction(Action.DELETE, workConfiguration, e)
     } finally {
       postClean.invoke()
     }
   }
 
+  private val markDurationTimer: LongTaskTimer = registry.longTaskTimer("swabbie.resources.mark.duration")
+  private val resourcesVisitedId: Id = registry.createId("swabbie.resources.visited")
+  private val resourcesExcludedId: Id = registry.createId("swabbie.resources.excluded")
+  private val resourceFailureId: Id = registry.createId("swabbie.resources.failures")
+  private val candidatesCountId: Id = registry.createId("swabbie.resources.candidatesCount")
+  private fun recordMarkMetrics(markerTimerId: Long,
+                                workConfiguration: WorkConfiguration,
+                                violationCounter: AtomicInteger,
+                                candidateCounter: AtomicInteger) {
+    markDurationTimer.stop(markerTimerId)
+    registry.gauge(
+      candidatesCountId.withTags(
+        "resourceType", workConfiguration.resourceType,
+        "configuration", workConfiguration.namespace,
+        "resourceHandler", javaClass.simpleName,
+        "violations", violationCounter.get().toString()
```

Comment:
I steered you wrong suggesting setting storing violation count as a tag - this is actually a bad idea as it results in a new timeseries for each unique value of `violationCounter`.

The easiest fix would be to store a separate metric for violations. A way to also store violation or other counts within `swabbie.resources.candidatesCount` without blowing up the backend would be to have something like a "status" tag with values like "marked" and "violations". Then update the gauge in separately for each status value, so that when `"status", "violations"`, you'd `set(violationCount.get())`. I think having a separate `swabbie.resources.markViolations` metric would make more sense though to someone browsing metrics without knowledge of the code.